### DISCUSSION TITLE
How can I make custom `Option` types work with graphql-kotlin?

### DISCUSSION THREAD

**Discussion by mainrs** [ExpediaGroup/graphql-kotlin#1905] [Q&A]
I am using ArrowKt, which defines a custom `arrow.option.Option<T>` type. I'd like to make it map to either `T`, if present, or `null` if no value is present.

Is there some way to make this library work with the ArrowKt types?

**Comment by dariuszkuc** [comment]
Hello 👋 
In general,  you would need to modify schema generation process so it can understand how to represent `Option` in the schema (see [schema generator hooks](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/customizing-schemas/generator-config#schemageneratorhooks)) and then you would also need to modify runtime process so you can correctly unwrap it (i.e. provide custom [DataFetcher](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt)).
